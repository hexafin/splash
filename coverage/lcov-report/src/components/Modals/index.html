<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Modals</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Modals
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.25% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>17/80</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">5.26% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/38</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/32</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.52% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>17/79</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="CardModal.js"><a href="CardModal.js.html">CardModal.js</a></td>
	<td data-value="10.87" class="pic low"><div class="chart"><div class="cover-fill" style="width: 10%;"></div><div class="cover-empty" style="width:90%;"></div></div></td>
	<td data-value="10.87" class="pct low">10.87%</td>
	<td data-value="46" class="abs low">5/46</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="5.56" class="pct low">5.56%</td>
	<td data-value="18" class="abs low">1/18</td>
	<td data-value="10.87" class="pct low">10.87%</td>
	<td data-value="46" class="abs low">5/46</td>
	</tr>

<tr>
	<td class="file medium" data-value="InfoModal.js"><a href="InfoModal.js.html">InfoModal.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	</tr>

<tr>
	<td class="file medium" data-value="ModalRoot.js"><a href="ModalRoot.js.html">ModalRoot.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	</tr>

<tr>
	<td class="file low" data-value="RaiseModal.js"><a href="RaiseModal.js.html">RaiseModal.js</a></td>
	<td data-value="15.79" class="pic low"><div class="chart"><div class="cover-fill" style="width: 15%;"></div><div class="cover-empty" style="width:85%;"></div></div></td>
	<td data-value="15.79" class="pct low">15.79%</td>
	<td data-value="19" class="abs low">3/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="9" class="abs low">1/9</td>
	<td data-value="15.79" class="pct low">15.79%</td>
	<td data-value="19" class="abs low">3/19</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Apr 17 2019 19:32:05 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
