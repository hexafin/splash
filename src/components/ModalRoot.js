import React from 'react'
import {
	Modal,
} from "react-native"
import { connect } from "react-redux";
import { bindActionCreators } from "redux"
import ViewTransactionModal from './ViewTransactionModal'
import ApproveTransactionModal from './ApproveTransactionModal'
import { hideModal } from '../redux/modal'

const MODAL_COMPONENTS = {
  'VIEW_TRANSACTION': ViewTransactionModal,
  'APPROVE_TRANSACTION': ApproveTransactionModal,
}

const ModalRoot = ({modalType, modalProps, hideModal}) => {
	if(!modalType) {
		return (null)
	} else {
		const Child = MODAL_COMPONENTS[modalType]
		return (
			<Modal
			    animationType="none"
		        transparent={true}
		        visible={true}>
		        <Child {...modalProps} hideModal={hideModal}/>
		    </Modal>
			)
	}
}

const mapStateToProps = state => state.modal

const mapDispatchToProps = dispatch => {
	return bindActionCreators(
		{
			hideModal
		},
		dispatch
	)
}

export default connect(mapStateToProps, mapDispatchToProps)(ModalRoot)
