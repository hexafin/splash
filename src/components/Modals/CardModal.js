import React, { Component } from "react";
import {
	Animated,
	Easing,
	View,
	TouchableWithoutFeedback,
	PanResponder,
	StyleSheet,
	Dimensions
} from "react-native"
import {colors} from "../../lib/colors"

const SCREEN_HEIGHT = Dimensions.get('window').height
const SCREEN_WIDTH = Dimensions.get('window').width

const CardModal = Child => {
 return class extends Component {
	  constructor(props) {
	    super(props)
	    this.state = {dismissed: false}
	    this.position = new Animated.ValueXY({x: SCREEN_WIDTH+70, y: 0})
		this.opacityAnimation = new Animated.Value(0.0)
	    this.rotate = this.position.x.interpolate({
	      inputRange: [-SCREEN_WIDTH / 2, 0, SCREEN_WIDTH / 2],
	      outputRange: ['-10deg', '0deg', '10deg'],
	      extrapolate: 'clamp'
	    })

	    this.rotateAndTranslate = {
	      transform: [{
	        rotate: this.rotate
	      },
	      ...this.position.getTranslateTransform()
	      ]
	    }
		this.handleClose = this.handleClose.bind(this)
	  }

	  componentWillMount() {
	    this.PanResponder = PanResponder.create({
		      onStartShouldSetPanResponder: (evt, gestureState) => true,
		      onPanResponderMove: (evt, gestureState) => {
		        this.position.setValue({ x: gestureState.dx, y: gestureState.dy })
		      },
		      onPanResponderRelease: (evt, gestureState) => {

		        if (gestureState.dx > 120) {
		          Animated.spring(this.position, {
		            toValue: { x: SCREEN_WIDTH + 100, y: gestureState.dy },
					useNativeDriver: true,
		          }).start()
				  Animated.spring(this.opacityAnimation, {
			        toValue: 0.0,
			      }).start(() => this.props.hideModal())
		        }
		        else if (gestureState.dx < -120) {
		          Animated.spring(this.position, {
		            toValue: { x: -SCREEN_WIDTH - 100, y: gestureState.dy },
					useNativeDriver: true,
		          }).start()
				  Animated.spring(this.opacityAnimation, {
			        toValue: 0.0,
			      }).start(() => this.props.hideModal())
		        }
		        else if (gestureState.dy > 120) {
		          Animated.spring(this.position, {
		            toValue: { x: gestureState.dx, y: SCREEN_HEIGHT + 100 },
					useNativeDriver: true,
		          }).start()
				  Animated.spring(this.opacityAnimation, {
			        toValue: 0.0,
			      }).start(() => this.props.hideModal())
		        }
		        else if (gestureState.dy < -120) {
		          Animated.spring(this.position, {
		            toValue: { x: gestureState.dx, y: -SCREEN_HEIGHT - 100 },
					useNativeDriver: true,
		          }).start()
				  Animated.spring(this.opacityAnimation, {
			        toValue: 0.0,
			      }).start(() => this.props.hideModal())
		        }
		        else {
		          Animated.spring(this.position, {
		            toValue: { x: 0, y: 0 },
		            friction: 4,
					useNativeDriver: true,
		          }).start()
		        }
		      },
		      onMoveShouldSetPanResponder: (evt, gestureState) => {
				    //return true if user is swiping, return false if it's a single click
	                return !(gestureState.dx === 0 && gestureState.dy === 0)                  
				}
		    })

	    	this.position.addListener(({x, y}) => {
		    	if (x >= SCREEN_WIDTH+80 || x <= -SCREEN_WIDTH-80 || y >= SCREEN_HEIGHT+80 || y <= -SCREEN_HEIGHT-80) {
		    			this.setState({dismissed: true}, () => {
			    			Animated.timing(this.position).stop()
		    			})
		    		}
	    	})
	  	}

	  	componentDidMount() {
			Animated.sequence([
				Animated.delay(10),
				Animated.parallel([
			        Animated.spring(this.position, {
			            toValue: { x: 0, y: 0 },
			            friction: 50,
						useNativeDriver: true,
			        }),
			        Animated.spring(this.opacityAnimation, {
			            toValue: 1.0,
			        }),
			    ])
			]).start();
	  	}

	  	componentWillUnmount() {
	  		this.position.removeAllListeners()
	  	}

	  	handleClose()  {
	      Animated.spring(this.position, {
	        toValue: { x: -SCREEN_WIDTH - 120, y: 0 },
	        friction: 20,
			useNativeDriver: true,
	      }).start()
		  Animated.spring(this.opacityAnimation, {
	        toValue: 0.0,
	      }).start(() => this.props.hideModal())
	  	}

		render() {

			let backgroundColor = 'rgba(67,68,167,0.32)'
			if (this.props.backgroundColor) {
				backgroundColor = this.props.backgroundColor
			}

			return (
			<TouchableWithoutFeedback onPress={this.handleClose} disabled={this.state.dismissed} pointerEvents={this.state.dismissed ? 'none' : 'auto'}>
				<Animated.View pointerEvents={this.state.dismissed ? 'none' : 'auto'}
							   style={[styles.container, {backgroundColor: this.opacityAnimation.interpolate(
																									{inputRange: [0, 1], outputRange: ['rgba(0,0,0,0)', backgroundColor]}
																									)}]}>
						{!this.state.dismissed && <Animated.View 
							{...this.PanResponder.panHandlers}
							style={[styles.popup, this.rotateAndTranslate]}>

							<Child {...this.props} handleClose={this.handleClose} />

						</Animated.View>}
				</Animated.View>
			</TouchableWithoutFeedback>
			)
		}
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: "rgba(0,0,0,0)",
		justifyContent: "center",
		alignItems: 'center',
		paddingHorizontal: 18,
		paddingVertical: 160,
	},
	popup: {
		position: 'absolute',
		backgroundColor: colors.white,
		borderRadius: 10,
		shadowOffset: {
			width: 0,
			height: 15,
		},
		shadowRadius: 25,
		shadowOpacity: 0.3
	},
})

export default CardModal
