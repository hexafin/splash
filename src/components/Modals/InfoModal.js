import React from "react"
import {
	View,
	StyleSheet,
	Text,
	TouchableOpacity,
	Image
} from "react-native";
import { colors } from "../../lib/colors"
import { icons, defaults } from "../../lib/styles";

const InfoModal = ({title, body, handleClose}) => {
	return (
		<View style={styles.container}>
			<View style={styles.header}>
			  <View />
              <Text style={styles.title}>{title}</Text>
	          <TouchableOpacity onPress={handleClose}>
	            <Image style={styles.closeButton} source={require('../../assets/icons/Xbutton.png')}/>
	          </TouchableOpacity>
	        </View>
	        <Text style={styles.content}>{body}</Text>
		</View>    							
	);
}

export default InfoModal;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		padding: 30,
	},
	header: {
	    flexDirection: 'row',
	    alignItems: 'center',
	    justifyContent: 'space-between',
	    paddingBottom: 17,
	},
	title: {
	    fontSize: 20,
	    fontWeight: "600",
	    color: colors.nearBlack,
	    alignSelf: 'center'
	},
	closeButton: {
	    alignSelf: 'center',
		height: 20,
		width: 20
	},
	content: {
		fontSize: 18,
		color: colors.nearBlack,
		textAlign: 'center',
		alignSelf: 'center'
	}
});