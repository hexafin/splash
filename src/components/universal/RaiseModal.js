import React, { Component } from "react";
import {
    StyleSheet,
    TouchableWithoutFeedback,
    TouchableOpacity,
    View,
    Text,
    Image,
    Easing,
	Animated,
} from "react-native"
import {colors} from "../../lib/colors"
import { icons } from "../../lib/styles";

const RaiseModal = Child => {
 return class extends Component {

		constructor(props) {
			super(props)
			this.animation = new Animated.Value(0.0)
			this.backgroundOpacity = new Animated.Value(0)
		}

		componentDidMount() {
			Animated.sequence([
				Animated.delay(10),
		        Animated.spring(this.animation, {
		            toValue: 1.0,
		        }),
			]).start();
		}

		render() {

			const modalTransform = () => {
				return {
					transform: [
						{
							translateY: this.animation.interpolate({
								inputRange: [0, 1],
								outputRange: [400, 0]
							})
						}
					]
				}
			}


			const dismiss = (success=false) => {
				Animated.timing(this.animation, {
					toValue: 0,
					duration: 200,
					easing: Easing.linear(),
				}).start(({finished}) => {
					if (finished) {
						if(success && this.props.successCallback) {
							this.props.successCallback()
						}
						this.props.hideModal()
						if (this.props.dismissCallback) {
							this.props.dismissCallback()
						}
					}
				})
			}

			return (
				<TouchableWithoutFeedback onPress={() => dismiss()}>
				<Animated.View style={[styles.container, {backgroundColor: this.animation.interpolate({inputRange: [0, 1], outputRange: ['rgba(0,0,0,0)', 'rgba(67,68,167,0.32)']})}]}>
					<View style={{ flexDirection: "row" }}>
						<TouchableWithoutFeedback onPress={() => {}}>
						<Animated.View style={[styles.popup, modalTransform()]}>

							<Child {...this.props} dismiss={dismiss} />

					</Animated.View>
					</TouchableWithoutFeedback>
				</View>
			</Animated.View>
			</TouchableWithoutFeedback>
			)
		}
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: "rgba(0,0,0,0)",
    justifyContent: 'flex-end'
	},
  popup: {
    flex: 1,
    flexDirection: "column",
    height: 400,
    alignItems: "center",
    borderTopLeftRadius: 15,
		borderTopRightRadius: 15,
    backgroundColor: colors.white
  },	
})

export default RaiseModal
