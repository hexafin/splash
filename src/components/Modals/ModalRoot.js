import React from 'react'
import {
	Modal,
} from "react-native"
import { connect } from "react-redux";
import { bindActionCreators } from "redux"
import ViewTransactionModal from '../ViewTransactionModal'
import ApproveTransactionModal from '../ApproveTransactionModal'
import ApproveCardModal from '../ApproveCardModal'
import DeleteContent from './content/DeleteContent'
import { hideModal } from '../../redux/modal'
import InfoModal from "./InfoModal"
import CardModal from "./CardModal"
import RaiseModal from "./RaiseModal"


const MODAL_COMPONENTS = {
  'VIEW_TRANSACTION': CardModal(ViewTransactionModal),
  'APPROVE_TRANSACTION': RaiseModal(ApproveTransactionModal),
  'APPROVE_CARD': RaiseModal(ApproveCardModal),
  'INFO': CardModal(InfoModal),
  'DELETE': CardModal(DeleteContent),
}

const ModalRoot = ({modalType, modalProps, hideModal}) => {
	if(!modalType) {
		return (null)
	} else {
		const Child = MODAL_COMPONENTS[modalType]
		return (
			<Modal
			    animationType="none"
		        transparent={true}
		        visible={true}>
		        <Child {...modalProps} hideModal={hideModal}/>
		    </Modal>
			)
	}
}

const mapStateToProps = state => state.modal

const mapDispatchToProps = dispatch => {
	return bindActionCreators(
		{
			hideModal
		},
		dispatch
	)
}

export default connect(mapStateToProps, mapDispatchToProps)(ModalRoot)
