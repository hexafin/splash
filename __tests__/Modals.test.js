import "react-native";
import React from "react";
import CardModal from "../src/components/Modals/CardModal";
import InfoModal from "../src/components/Modals/InfoModal";
import { ModalRoot } from "../src/components/Modals/ModalRoot";
import RaiseModal from "../src/components/Modals/RaiseModal";

it("Card Modal renders correctly", () => {
	const snap = shallow(<CardModal />);
	expect(snap).toMatchSnapshot();
});

it("Info Modal renders correctly", () => {
	const snap = shallow(<InfoModal />);
	expect(snap).toMatchSnapshot();
});

it("Modal Root renders correctly", () => {
	const snap = shallow(<ModalRoot />);
	expect(snap).toMatchSnapshot();
});

it("Raise Modal renders correctly", () => {
	const snap = shallow(<RaiseModal />);
	expect(snap).toMatchSnapshot();
});
