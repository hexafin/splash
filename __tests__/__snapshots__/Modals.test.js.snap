// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Modal renders correctly 1`] = `<Component />`;

exports[`Info Modal renders correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "padding": 30,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 17,
      }
    }
  >
    <Component
      style={
        Object {
          "height": 20,
          "margin": 8,
          "width": 20,
        }
      }
    />
    <Component
      style={
        Object {
          "color": "#333333",
          "fontSize": 20,
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <Component
        source={
          Object {
            "testUri": "../../../src/assets/icons/Xbutton.png",
          }
        }
        style={
          Object {
            "height": 20,
            "margin": 8,
            "width": 20,
          }
        }
      />
    </TouchableOpacity>
  </Component>
  <Component
    style={
      Object {
        "alignSelf": "center",
        "color": "#333333",
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  />
</Component>
`;

exports[`Modal Root renders correctly 1`] = `""`;

exports[`Raise Modal renders correctly 1`] = `<Component />`;
